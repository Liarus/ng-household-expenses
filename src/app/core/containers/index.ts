export * from './app';
export * from './page-not-found';
