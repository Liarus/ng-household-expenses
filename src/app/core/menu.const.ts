import { MenuItem } from './models/menuItem.model';

export const Menu: MenuItem[] = [
    {
        url: '/app-households',
        title: 'Households',
        icon: 'home',
        permissions: ['CanSeeHouseholds']
    },
    {
        url: '/app-expenses',
        title: 'Expenses',
        icon: 'payment',
        permissions: ['CanSeeExpenses']
    },
    {
        url: '/app-savings',
        title: 'Savings',
        icon: 'drafts',
        permissions: ['CanSeeSavings']
    },
    {
        url: '/app-settings',
        title: 'Settings',
        icon: 'settings',
        permissions: ['CanSeeSettings']
    },
    {
        url: '/app-users',
        title: 'Users',
        icon: 'group',
        permissions: ['CanSeeUsers']
    }
];
