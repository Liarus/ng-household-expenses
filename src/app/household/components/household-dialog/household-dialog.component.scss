.mat-expansion-panel-header {
  padding: 0;
  padding-right: 0.5em;
}

.mat-expansion-panel {
  box-shadow: none;
}

::ng-deep .mat-expansion-panel-body {
    padding: 0 !important;
}

::ng-deep .cdk-overlay-backdrop {
    pointer-events: none;
}
