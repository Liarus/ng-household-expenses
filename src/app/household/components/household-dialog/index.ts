export * from './household-dialog.component';
