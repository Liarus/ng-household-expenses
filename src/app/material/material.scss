.mat-bottom-sheet-container {
  padding: 0 !important;
}

.mat-table {
  mat-row, mat-header-row, mat-footer-row {
    padding-left: 24px;
    padding-right: 24px;
  }
  
  mat-cell:first-child, mat-footer-cell:first-child, mat-header-cell:first-child {
    padding-left: 0;
  }
  
  mat-cell:last-child, mat-footer-cell:last-child, mat-header-cell:last-child {
    padding-right: 0;
  }
}

mat-expansion-panel {
  mat-expansion-panel-header {
    height: 48px; // height may be different for you
  }
  .mat-expansion-panel-content {
    height: 0;
  }
  &.mat-expanded {
    mat-expansion-panel-header {
      height: 64px; // height may be different for you
    }
    .mat-expansion-panel-content {
      height: auto;
    }
  }
}