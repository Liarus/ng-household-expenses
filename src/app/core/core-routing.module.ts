import { Routes, RouterModule } from '@angular/router';

import { LayoutComponent } from './containers';
import { NgModule } from '@angular/core';

export const routes: Routes = [
    {
        path: 'root',
        component: LayoutComponent
    }
];

@NgModule({
    imports: [RouterModule.forRoot(routes)],
    exports: [RouterModule]
  })
export class CoreRoutingModule { }
