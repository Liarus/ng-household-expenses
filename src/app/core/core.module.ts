import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NgModule} from '@angular/core';

const PROVIDERS = [
];

@NgModule({
  imports: [
    CommonModule,
    RouterModule
  ]
})
export class CoreModule {}
