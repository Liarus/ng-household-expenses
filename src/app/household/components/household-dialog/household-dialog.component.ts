import { Component, Inject, OnInit } from '@angular/core';
import { FormGroup, FormControl, Validators } from '@angular/forms';
import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';

import { HouseholdDialog } from '../../models/householdDialog.model';

const fg = dataItem => new FormGroup({
  id: new FormControl(dataItem.id),
  version: new FormControl(dataItem.version),
  userId: new FormControl(dataItem.userId),
  name: new FormControl(dataItem.name, [Validators.required, Validators.maxLength(255)]),
  symbol: new FormControl(dataItem.symbol, [Validators.maxLength(255)]),
  description: new FormControl(dataItem.description, [Validators.maxLength(255)]),
  street: new FormControl(dataItem.street, [Validators.required, Validators.maxLength(255)]),
  city: new FormControl(dataItem.city, [Validators.required, Validators.maxLength(255)]),
  country: new FormControl(dataItem.country, [Validators.required, Validators.maxLength(255)]),
  zipCode: new FormControl(dataItem.zipCode, [Validators.required, Validators.maxLength(255)])
});

@Component({
  selector: 'app-household-dialog',
  templateUrl: './household-dialog.component.html',
  styleUrls: ['./household-dialog.component.scss']
})

export class HouseholdDialogComponent {
  form: FormGroup;

  constructor(public dialogRef: MatDialogRef<HouseholdDialogComponent>,
    @Inject(MAT_DIALOG_DATA) public data: HouseholdDialog) {
      this.form = fg(Object.assign({}, data.household, { userId: data.userId }));
    }

  getAddressErrorMessage() {
    return (this.form.controls['street'].touched && this.form.controls['street'].errors)
      || (this.form.controls['city'].touched && this.form.controls['city'].errors)
      || (this.form.controls['country'].touched && this.form.controls['country'].errors)
      || (this.form.controls['zipCode'].touched && this.form.controls['zipCode'].errors)
      ? 'Please fill address correctly' : '';
  }

  onCancelClick() {
    this.dialogRef.close();
  }
}
