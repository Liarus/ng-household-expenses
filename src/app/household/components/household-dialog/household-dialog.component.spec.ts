import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { HouseholdDialogComponent } from './household-dialog.component';
import { MaterialModule } from '../../../material/material.module';
import { HouseholdDialog } from '../../models/householdDialog.model';
import { TEST_DATA } from '../../../shared/tests/test-data';

describe('HouseholdDialogComponent', () => {
  let component: HouseholdDialogComponent;
  let fixture: ComponentFixture<HouseholdDialogComponent>;
  let matDialogRef: MatDialogRef<HouseholdDialogComponent>;

  const matDialogData: HouseholdDialog = {
    userId: TEST_DATA.auth.userId,
    household: TEST_DATA.household.household
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        NoopAnimationsModule,
        ReactiveFormsModule
      ],
      declarations: [ HouseholdDialogComponent ],
      providers: [
        { provide: MatDialogRef, useValue: {
          close: jest.fn()
        }
       },
        { provide: MAT_DIALOG_DATA, useValue: matDialogData }
    ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    matDialogRef = TestBed.get(MatDialogRef);
    spyOn(matDialogRef, 'close');
    fixture = TestBed.createComponent(HouseholdDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should should match snapshot if address invalid', () => {
    component.form.setValue({
      userId: TEST_DATA.auth.userId,
      id: TEST_DATA.household.household.id,
      name: TEST_DATA.household.household.name,
      symbol: TEST_DATA.household.household.symbol,
      description: TEST_DATA.household.household.description,
      street: '',
      city: null,
      country: TEST_DATA.household.household.country,
      zipCode: TEST_DATA.household.household.zipCode,
      version: TEST_DATA.household.household.version
    });

    component.form.controls['city'].markAsTouched();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should close dialog', () => {
    component.onCancelClick();

    expect(matDialogRef.close).toHaveBeenCalled();
  });
});
