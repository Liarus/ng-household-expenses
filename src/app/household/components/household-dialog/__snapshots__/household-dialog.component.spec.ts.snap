// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseholdDialogComponent should should match snapshot if address invalid 1`] = `
<app-household-dialog
  data={[Function Object]}
  dialogRef={[Function Object]}
  form={[Function FormGroup]}
>
  <h1
    class="mat-dialog-title"
    id="mat-dialog-title-1"
    mat-dialog-title=""
  >
    Create Household
  </h1><form
    class="ng-pristine ng-touched ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <mat-dialog-content
      class="mat-dialog-content"
      fxlayout="column"
    >
      <div
        fxlayout="row"
        fxlayout.lt-md="column"
        fxlayoutgap="20px"
        fxlayoutgap.lt-md="0px"
      >
        <mat-form-field
          class="mat-form-field ng-tns-c4-0 mat-accent mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-should-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
          color="accent"
          fxflex=""
          ng-reflect-color="accent"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                  id="mat-input-7"
                  matinput=""
                  ng-reflect-form="[object Object]"
                  ng-reflect-placeholder="*Name"
                  placeholder="*Name"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-7"
                    class="mat-form-field-label ng-tns-c4-0 mat-accent ng-star-inserted"
                    for="mat-input-7"
                    id="mat-form-field-label-15"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="false"
                  >
                    
                    
                    *Name
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c4-0 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple mat-accent"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c4-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          class="mat-form-field ng-tns-c4-1 mat-accent mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-should-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
          color="accent"
          fxflex=""
          ng-reflect-color="accent"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                  id="mat-input-8"
                  matinput=""
                  ng-reflect-form="[object Object]"
                  ng-reflect-placeholder="Symbol"
                  placeholder="Symbol"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-8"
                    class="mat-form-field-label ng-tns-c4-1 mat-accent ng-star-inserted"
                    for="mat-input-8"
                    id="mat-form-field-label-17"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="false"
                  >
                    
                    
                    Symbol
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c4-1 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple mat-accent"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c4-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
      </div>
      <mat-form-field
        class="mat-form-field ng-tns-c4-2 mat-accent mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-should-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
        color="accent"
        fxflex=""
        ng-reflect-color="accent"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-infix"
            >
              <textarea
                aria-invalid="false"
                aria-required="false"
                class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                id="mat-input-9"
                matinput=""
                ng-reflect-form="[object Object]"
                ng-reflect-placeholder="Description"
                placeholder="Description"
              />
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  aria-owns="mat-input-9"
                  class="mat-form-field-label ng-tns-c4-2 mat-accent ng-star-inserted"
                  for="mat-input-9"
                  id="mat-form-field-label-19"
                  ng-reflect-disabled="true"
                  ng-reflect-ng-switch="false"
                >
                  
                  
                  Description
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-underline ng-tns-c4-2 ng-star-inserted"
          >
            <span
              class="mat-form-field-ripple mat-accent"
            />
          </div>
          <div
            class="mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-tns-c4-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      
      <mat-error
        class="mat-error ng-star-inserted"
        id="mat-error-0"
        role="alert"
      >
        Please fill address correctly
      </mat-error>
      <mat-expansion-panel
        class="mat-expansion-panel ng-tns-c5-3 _mat-animation-noopable"
      >
        <mat-expansion-panel-header
          aria-controls="cdk-accordion-child-1"
          aria-disabled="false"
          aria-expanded="false"
          class="mat-expansion-panel-header ng-tns-c6-4 ng-trigger ng-trigger-expansionHeight ng-star-inserted"
          id="mat-expansion-panel-header-1"
          role="button"
          style="height:48px;0:height;"
          tabindex="0"
        >
          <span
            class="mat-content"
          >
            <mat-panel-title
              class="mat-expansion-panel-header-title"
            >
              Address
            </mat-panel-title>
            <mat-panel-description
              class="mat-expansion-panel-header-description"
            >
              Fill company main address
            </mat-panel-description>
          </span>
          
          <span
            class="mat-expansion-indicator ng-tns-c6-4 ng-trigger ng-trigger-indicatorRotate ng-star-inserted"
            style="transform:rotate(0deg);0:transform;"
          />
        </mat-expansion-panel-header>
        <div
          aria-labelledby="mat-expansion-panel-header-1"
          class="mat-expansion-panel-content ng-trigger ng-trigger-bodyExpansion"
          id="cdk-accordion-child-1"
          role="region"
          style="height:0px;visibility:hidden;0:height;1:visibility;"
        >
          <div
            class="mat-expansion-panel-body"
          >
            <div
              fxlayout="column"
            >
              <mat-form-field
                class="mat-form-field ng-tns-c4-5 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine _mat-animation-noopable ng-invalid"
                fxflex=""
              >
                <div
                  class="mat-form-field-wrapper"
                >
                  <div
                    class="mat-form-field-flex"
                  >
                    
                    
                    <div
                      class="mat-form-field-infix"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="false"
                        class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                        id="mat-input-10"
                        matinput=""
                        ng-reflect-form="[object Object]"
                        ng-reflect-placeholder="*Street"
                        placeholder="*Street"
                      />
                      <span
                        class="mat-form-field-label-wrapper"
                      >
                        
                        <label
                          aria-owns="mat-input-10"
                          class="mat-form-field-label ng-tns-c4-5 ng-star-inserted mat-empty mat-form-field-empty"
                          for="mat-input-10"
                          id="mat-form-field-label-21"
                          ng-reflect-disabled="true"
                          ng-reflect-ng-switch="false"
                          style=""
                        >
                          
                          
                          *Street
                          
                          
                        </label>
                      </span>
                    </div>
                    
                  </div>
                  
                  <div
                    class="mat-form-field-underline ng-tns-c4-5 ng-star-inserted"
                    style=""
                  >
                    <span
                      class="mat-form-field-ripple"
                    />
                  </div>
                  <div
                    class="mat-form-field-subscript-wrapper"
                    ng-reflect-ng-switch="hint"
                  >
                    
                    
                    <div
                      class="mat-form-field-hint-wrapper ng-tns-c4-5 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                      style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                    >
                      
                      <div
                        class="mat-form-field-hint-spacer"
                      />
                    </div>
                  </div>
                </div>
              </mat-form-field>
              <div
                fxlayout="row"
                fxlayout.lt-md="column"
                fxlayoutgap="20px"
                fxlayoutgap.lt-md="0px"
              >
                <mat-form-field
                  class="mat-form-field ng-tns-c4-6 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-should-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
                  fxflex=""
                >
                  <div
                    class="mat-form-field-wrapper"
                  >
                    <div
                      class="mat-form-field-flex"
                    >
                      
                      
                      <div
                        class="mat-form-field-infix"
                      >
                        <input
                          aria-invalid="false"
                          aria-required="false"
                          class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                          id="mat-input-11"
                          matinput=""
                          ng-reflect-form="[object Object]"
                          ng-reflect-placeholder="*Zip code"
                          placeholder="*Zip code"
                        />
                        <span
                          class="mat-form-field-label-wrapper"
                        >
                          
                          <label
                            aria-owns="mat-input-11"
                            class="mat-form-field-label ng-tns-c4-6 ng-star-inserted"
                            for="mat-input-11"
                            id="mat-form-field-label-23"
                            ng-reflect-disabled="true"
                            ng-reflect-ng-switch="false"
                            style=""
                          >
                            
                            
                            *Zip code
                            
                            
                          </label>
                        </span>
                      </div>
                      
                    </div>
                    
                    <div
                      class="mat-form-field-underline ng-tns-c4-6 ng-star-inserted"
                      style=""
                    >
                      <span
                        class="mat-form-field-ripple"
                      />
                    </div>
                    <div
                      class="mat-form-field-subscript-wrapper"
                      ng-reflect-ng-switch="hint"
                    >
                      
                      
                      <div
                        class="mat-form-field-hint-wrapper ng-tns-c4-6 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                        style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                      >
                        
                        <div
                          class="mat-form-field-hint-spacer"
                        />
                      </div>
                    </div>
                  </div>
                </mat-form-field>
                <mat-form-field
                  class="mat-form-field ng-tns-c4-7 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-pristine _mat-animation-noopable mat-form-field-invalid ng-touched ng-invalid"
                  fxflex=""
                >
                  <div
                    class="mat-form-field-wrapper"
                  >
                    <div
                      class="mat-form-field-flex"
                    >
                      
                      
                      <div
                        class="mat-form-field-infix"
                      >
                        <input
                          aria-invalid="true"
                          aria-required="false"
                          class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-pristine ng-touched ng-invalid"
                          id="mat-input-12"
                          matinput=""
                          ng-reflect-form="[object Object]"
                          ng-reflect-placeholder="*City"
                          placeholder="*City"
                        />
                        <span
                          class="mat-form-field-label-wrapper"
                        >
                          
                          <label
                            aria-owns="mat-input-12"
                            class="mat-form-field-label ng-tns-c4-7 ng-star-inserted mat-empty mat-form-field-empty"
                            for="mat-input-12"
                            id="mat-form-field-label-25"
                            ng-reflect-disabled="true"
                            ng-reflect-ng-switch="false"
                            style=""
                          >
                            
                            
                            *City
                            
                            
                          </label>
                        </span>
                      </div>
                      
                    </div>
                    
                    <div
                      class="mat-form-field-underline ng-tns-c4-7 ng-star-inserted"
                      style=""
                    >
                      <span
                        class="mat-form-field-ripple"
                      />
                    </div>
                    <div
                      class="mat-form-field-subscript-wrapper"
                      ng-reflect-ng-switch="hint"
                    >
                      
                      
                      <div
                        class="mat-form-field-hint-wrapper ng-tns-c4-7 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                        style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                      >
                        
                        <div
                          class="mat-form-field-hint-spacer"
                        />
                      </div>
                    </div>
                  </div>
                </mat-form-field>
              </div>
              <mat-form-field
                class="mat-form-field ng-tns-c4-8 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-should-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
                fxflex=""
              >
                <div
                  class="mat-form-field-wrapper"
                >
                  <div
                    class="mat-form-field-flex"
                  >
                    
                    
                    <div
                      class="mat-form-field-infix"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="false"
                        class="form-control mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                        id="mat-input-13"
                        matinput=""
                        ng-reflect-form="[object Object]"
                        ng-reflect-placeholder="*Country"
                        placeholder="*Country"
                      />
                      <span
                        class="mat-form-field-label-wrapper"
                      >
                        
                        <label
                          aria-owns="mat-input-13"
                          class="mat-form-field-label ng-tns-c4-8 ng-star-inserted"
                          for="mat-input-13"
                          id="mat-form-field-label-27"
                          ng-reflect-disabled="true"
                          ng-reflect-ng-switch="false"
                          style=""
                        >
                          
                          
                          *Country
                          
                          
                        </label>
                      </span>
                    </div>
                    
                  </div>
                  
                  <div
                    class="mat-form-field-underline ng-tns-c4-8 ng-star-inserted"
                    style=""
                  >
                    <span
                      class="mat-form-field-ripple"
                    />
                  </div>
                  <div
                    class="mat-form-field-subscript-wrapper"
                    ng-reflect-ng-switch="hint"
                  >
                    
                    
                    <div
                      class="mat-form-field-hint-wrapper ng-tns-c4-8 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                      style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                    >
                      
                      <div
                        class="mat-form-field-hint-spacer"
                      />
                    </div>
                  </div>
                </div>
              </mat-form-field>
            </div>
            
          </div>
        </div>
      </mat-expansion-panel>
    </mat-dialog-content>
    <div
      class="mat-dialog-actions"
      mat-dialog-actions=""
    >
      <button
        cdkfocusinitial=""
        class="mat-button _mat-animation-noopable"
        mat-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          Cancel
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
      <button
        aria-label="Close dialog"
        class="mat-button _mat-animation-noopable"
        disabled=""
        mat-button=""
        ng-reflect-dialog-result="[object Object]"
        ng-reflect-disabled="true"
        type="button"
      >
        <span
          class="mat-button-wrapper"
        >
          Ok
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="true"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </form>
</app-household-dialog>
`;
